internal/proto/scan.go

```diff
@@ -105,3 +106,63 @@ func Scan(b []byte, v interface{}) error {
 			"redis: can't unmarshal %T (consider implementing BinaryUnmarshaler)", v)
 	}
 }
+
+
+// Scan a string slice into a custom container
+// Example:
+// var container []YourStruct; ScanSlice([]string{""},&container)
+// var container []*YourStruct; ScanSlice([]string{""},&container)
+func ScanSlice(sSlice []string, container interface{}) error {
+	val := reflect.ValueOf(container)
```

Comment:
You can/should immediately check if value is valid here. I think you can use [this code](https://github.com/go-pg/pg/blob/v5/types/scan.go#L52-L62) I wrote for another project. Then you should be able to get rid of `errTyp` flag.